native/mod_manager/node.c

```diff
@@ -229,30 +244,48 @@ apr_status_t find_node(mem_t *s, nodeinfo_t **node, const char *route)
 
     strncpy(ou.mess.JVMRoute, route, sizeof(ou.mess.JVMRoute));
     ou.mess.JVMRoute[sizeof(ou.mess.JVMRoute) - 1] = '\0';
-    *node = &ou;
-    rv = s->storage->ap_slotmem_do(s->slotmem, loc_read_node, node, s->p);
+    rv = s->storage->doall(s->slotmem, loc_read_node, &ou, s->p);
+    if (rv == APR_SUCCESS)
+        return APR_NOTFOUND;
+    if (rv == APR_EEXIST) {
+        rv = s->storage->dptr(s->slotmem, ou.mess.id, (void **)node);
+    }
     return rv;
 }
 
 /*
  * get the ids for the used (not free) nodes in the table
  * @param pointer to the shared table.
  * @param ids array of int to store the used id (must be big enough).
- * @return number of node existing or -1 if error.
+ * @return number of node existing or 0 if error.
  */
+static apr_status_t loc_get_id(void *mem, void *data, apr_pool_t *pool)
+{
+    struct counter *count = (struct counter *)data;
+    nodeinfo_t *ou = (nodeinfo_t *)mem;
+    *count->values = ou->mess.id;
+    count->values++;
+    count->count++;
+    return APR_SUCCESS;
+}
```

Comment:
The description is not ok. The function returns status and also `APR_SUCCESS` may equal to `0` but should not signal an error.